pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -326,8 +373,13 @@ static __always_inline void parse_frame(struct __sk_buff *skb, skb_info_t *skb_i
         bpf_map_delete_elem(&http2_in_flight, &http2_ctx->http2_stream_key);
         return;
     }
-    if (is_rst || ((current_frame->flags & HTTP2_END_OF_STREAM) == HTTP2_END_OF_STREAM)) {
-        handle_end_of_stream(current_stream, &http2_ctx->http2_stream_key);
+
+    if (current_frame->type == kRSTStreamFrame) {
+        __sync_fetch_and_add(&http2_tel->end_of_stream_rst, 1);
+        handle_end_of_stream(current_stream, &http2_ctx->http2_stream_key, http2_tel);
```

Comment:
no need to call handle_end_of_stream twice
we're using `__always_inline`, so you're duplicating the number of instructions and increasing the complexity of the program by that